gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/ManagedController.java

```diff
@@ -135,6 +136,30 @@ DeliveryConfig upsertManifest(@RequestBody DeliveryConfig manifest) {
     return keelService.diffManifest(manifest);
   }
 
+  @ApiOperation(
+      value = "List up-to {limit} current constraint states for an environment",
+      response = ConstraintState.class)
+  @GetMapping(path = "/delivery-configs/{name}/environment/{environment}/constraints")
+  List<ConstraintState> getConstraintState(
+      @PathVariable("name") String name,
+      @PathVariable("environment") String environment,
+      @RequestParam(value = "limit", required = false, defaultValue = "10") String limit) {
+    return keelService.getConstraintState(name, environment, Integer.valueOf(limit));
+  }
+
+  @ApiOperation(value = "Update the status of an environment constraint")
+  @PostMapping(path = "/delivery-configs/{name}/environment/{environment}/constraint/{type}")
+  void updateConstraintStatus(
+      @PathVariable("name") String name,
+      @PathVariable("environment") String environment,
+      @PathVariable("type") String type,
+      @RequestParam(value = "artifactVersion") String artifactVersion,
+      @RequestParam(value = "status") String status,
+      @RequestParam(value = "comment", required = false, defaultValue = "") String comment) {
+    keelService.updateConstraintStatus(
+        name, environment, type, artifactVersion, status, "".equals(comment) ? null : comment);
```

Comment:
Wouldn't `comment` default to `null` if you removed the `defaultValue` from the annotation, in which case you wouldn't need to convert here?